dll/win32/shell32/shlexec.cpp

```diff
@@ -458,6 +458,132 @@ static HRESULT SHELL_GetPathFromIDListForExecuteW(LPCITEMIDLIST pidl, LPWSTR psz
     return hr;
 }
 
+static HWND SHELL_GetUsableDialogOwner(HWND hwnd)
+{
+    // Explicitly block the shell desktop listview from becoming the owner (IContextMenu calling ShellExecute)
+    HWND progman = GetShellWindow();
+    if (hwnd && IsWindowVisible(hwnd) && hwnd != GetDesktopWindow() &&
+        hwnd != progman && !IsChild(progman, hwnd))
+    {
+        return hwnd;
+    }
+    return NULL;
+}
+
+/*************************************************************************
+ *    SHELL_PromptAndRunProcessAs [Internal]
+ */
+typedef struct {
+    WCHAR NameBuffer[MAX_PATH], Pass[MAX_PATH];
+    LPWSTR Name, Domain;
+    BOOL Safer;
+} RUNASDLGDATA;
+
+static INT_PTR CALLBACK RunAsDlgProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    RUNASDLGDATA *pData = (RUNASDLGDATA *)GetWindowLongPtr(hwnd, DWLP_USER);
```

Comment:
I guess that instead of having `RUNASDLGDATA *` copy-pasted everytime, you could add, in the typedef above, `*PRUNASDLGDATA`, and use that latter `PRUNASDLGDATA` as the type here and below.